### DISCUSSION TITLE
Just lurking around your new stuff

### DISCUSSION THREAD

**Discussion by Hax0rgurl** [rslay/ZeroChat#147] [General]
and wanted to say hi.  Noticed that barebones OS?  Does it work?  Gonna grab it and check it out.  
miss our POC FIU days. 
Hope all is well
T

**Comment by rslay** [comment]
Hey @Hax0rgurl! At the moment it works on some windows machines, but the barebones OS is very very simple. It just prints text after running the bootloader.

I was hoping to eventually get the stdlib for C in there so I can do things like `printf()` and that sort of thing, without having to stress about re-implementing it myself, and make an entire OS that revolves around running a single C file very efficiently with no other overhead - no windows, no linux, just a bootloader and code. The idea was to build something else with this, like a 3D game that runs stupid-fast.

Also, there's chat.justhack.in, running Zerochat, which is a live chat site without javascript or cookies or anything, just hacky iframes loading up everything necessary to run a chat and stream HTML/CSS. It's mostly finished now.

Everything is well on my end, and on my way to become financially independent soon enough.

Let me know if you got some projects yourself that you'd want help with building out!

**Reply by Hax0rgurl** [reply]
So how would you resolve the internet dilemma? In your ideal scenario would the game be completely offline? I mostly play offline, that's why I ask. I don't like EA collecting data on how I play. That sounds sick though.  Just pure barebones to play games sounds like a dream.  I have a laptop perfect for that sort of thing too. As long as it ideally supports the .Net Framework. Did you abandon the project or you still working on it?  I'd be down to lend a hand. I'm a lot more familiar now with the ways of Windows and it's interior functions than when we first met. Sounds like a worthy cause :)

**Reply by rslay** [reply]
Well, it'd require a lot of windows dependencies, and the OS would have quite literally zero software dependencies or drivers, it's just a raw OS. But instead of coding a file in C# for the OS to run after it's finished working, you could code something in C++, not too different. 

I haven't worked on it in a bit because it was very challenging getting the OS to compile into an ISO that would run raw on a computer like a USB - but, I was following [this guide](https://wiki.osdev.org/Babystep1) on osdev wiki.

Unfortunately, while it looks like it's straightforward and just a few steps - it's actually rather challenging getting everything working and set up properly.

I would actually be **very appreciative** if you tried to see if you can run it yourself. I made some instructions to follow and got everything that you should need for the OS to run in a virtual machine on Windows, explained in the readme of the project.

Think you can try your hand [at this](https://github.com/rslay/osdev-barebones/)?

**Reply by Hax0rgurl** [reply]
I'll give it a go this week and let you know. I have virtualbox on my macbook that I can run it on. That's where I run all my VMs.

**Reply by rslay** [reply]
Thanks to you, I finally went back to trying to get this up and running and figured out what was taking me weeks of confusion and struggling in the span of just 2 hours, I think I got every step necessary carefully explained in the readme now and the issues I was having before might have finally been fixed. 

If you run it in a Windows VM, that should work. Just let me know if you run into trouble and show me the error, and I'll try my best to see what's wrong.
